### DISCUSSION TITLE
[Security Feature: allow only connections to whitelisted Rustdesk Numbers]

### DISCUSSION THREAD

**Discussion by oliuo1** [rustdesk/rustdesk#3130] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

As a Rustdesk Server maitnainer i only want to have connections between MY COMPUTERS (identified via the rustdesk numbers) and everbody else who is on my server. I DONT want to give everybody else on my server the possibility to establish a connection with someone else - ONLY TO MY COMPUTERS

### Who would benefit from such a feature?

all users who have their own Rustdesk Server and support a medium user group via Rustdesk
even if your rustdesk server is used not only by all persons you support but also from "others" --> THEY (the persons you are supporting AND everybody else) ARE NOT ALLOWED to establish connections between each other

Example 1: 
I support User A from Company A
I support User B from Compaby A
User A and B sitting in the same office and the know the Rustdesk Numbers of each other

-> now they can use my Rustdesk Server (and my bandwith and so on) to connect to each other.

Example 2: my Rustdesk Server and Key gets public for whatever reason. now everybody who is connecting can use my server to establish connections to anybody else on my server (if they change Rustdesk numbers/know the numbers)
-> thats not good!

Solution:
If someone tries to establish a connection to another computer which is not whitelisted a message could appear: "This is a private Helpdesk Server. Only connections to computers which are legitimated are allowed!" and no connection gets established. So a connection to the rustdesk server is okay (cause that cant be disallowed). But a user cant do anything after connecting. As soon as he wants to connect to another computer which is NOT MINE (which hasnt been whitelisted via the Rustdesk Number) he gets the message and no connection will be made!

**Comment by alirezaalavi87** [comment]
I have this exact issue. If keys can be just redistributed, there's no point in having keys and your servers are never private.

**Comment by n1smithy** [comment]
I'm working as a IT supporter and I thouht about the exact same problem!

I don't want everyone to use my infrastructure without permission for security reasons. Would not be cool if scammers use my server for criminal stuff. And exspecially I'm not very comfortable with a possible visit by the cops taking my server as a criminal tool for research.
